File: frontend/src/types/project.ts
export interface ProjectAgent {
  agent_type: string;
  name: string;
  icon: string;
  status: string;
  task_completed: boolean;
}

export interface Project {
  id: string;
  name: string;
  description: string | null;
  status: string;
  agents: ProjectAgent[];
  agents_count: number;
  created_at: string;
}

export interface RecommendedAgent {
  type: string;
  reason: string;
}

export interface ProjectCreateResult {
  id: string;
  name: string;
  description: string | null;
  status: string;
  recommended_agents: RecommendedAgent[];
  created_at: string;
}

export interface AgentTemplate {
  id: string;
  type: string;
  name: string;
  description: string;
  icon: string;
  base_price: number;
  capabilities: string[];
}

File: backend/app/schemas/project.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.enums import AgentType, ProjectStatus


class ProjectCreate(BaseModel):
    description: str


class ProjectUpdate(BaseModel):
    name: str | None = None
    status: ProjectStatus | None = None


class AgentInfo(BaseModel):
    agent_type: AgentType
    name: str
    icon: str
    status: str
    task_completed: bool = False

    model_config = {"from_attributes": True}


class RecommendedAgentInfo(BaseModel):
    type: str
    reason: str


class ProjectResponse(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None
    status: ProjectStatus
    agents: list[AgentInfo] = []
    agents_count: int = 0
    created_at: datetime

    model_config = {"from_attributes": True}


class ProjectCreateResponse(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None
    status: ProjectStatus
    recommended_agents: list[RecommendedAgentInfo] = []
    created_at: datetime


class AddAgentRequest(BaseModel):
    agent_type: AgentType
